Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs

```diff
@@ -257,61 +257,149 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
                 context.Debug($"GetLruPages start with contentHashesWithInfo.Count={contentHashesWithInfo.Count}, firstAge={first.Age}, lastAge={last.Age}");
             }
 
+            var operationContext = new OperationContext(context);
+
             var pageSize = _configuration.EvictionWindowSize;
+            var subPageSize = Math.Max(1, pageSize / 10);
 
-            // Priority queue orders by least first. So we compare by last access time to get the least last access time (i.e. oldest) first.
-            var priorityQueue = new PriorityQueue<ContentHashWithLastAccessTimeAndReplicaCount>(
-                pageSize * 2,
-                // Assume that EffectiveLastAccessTime will always have a value.
-                Comparer<ContentHashWithLastAccessTimeAndReplicaCount>.Create((c1, c2) => c1.EffectiveLastAccessTime.Value.CompareTo(c2.EffectiveLastAccessTime.Value)));
+            // Assume that EffectiveLastAccessTime will always have a value.
+            var comparer = Comparer<ContentHashWithLastAccessTimeAndReplicaCount>.Create((c1, c2) => c1.EffectiveLastAccessTime.Value.CompareTo(c2.EffectiveLastAccessTime.Value));
 
-            var operationContext = new OperationContext(context);
+            Func<List<ContentHashWithLastAccessTimeAndReplicaCount>, IEnumerable<ContentHashWithLastAccessTimeAndReplicaCount>> query =
+                page => _localLocationStore.GetEffectiveLastAccessTimes(operationContext, page.SelectList(v => new ContentHashWithLastAccessTime(v.ContentHash, v.LastAccessTime))).ThrowIfFailure();
 
-            var subPageSize = Math.Max(1, pageSize / 10);
+            // We make sure that we we select a set of the newer content, to ensure that we at least look at newer content to see if it should be
+            // evicted first due to having a high number of replicas. We do this by looking at the start as well as at middle of the list.
+            var localOldest = QueryAndOrderInPages(contentHashesWithInfo.Take(contentHashesWithInfo.Count / 2), pageSize, comparer, query);
+            var localMid = QueryAndOrderInPages(contentHashesWithInfo.Skip(contentHashesWithInfo.Count / 2), pageSize, comparer, query);

```

Comment:
>Skip [](start = 70, length = 4)

Skip isn't optimized for lists. Maybe just have QueryAndOrderInPages take a list with start and end offset.